src/devices/srsmith_pool_srs_2c_tx.c

```diff
@@ -0,0 +1,175 @@
+        bitbuffer_extract_bytes(bitbuffer, 0, 8, b, 18 * 8); // shift in case first bit was received properly
+    }
+    else {
+        return DECODE_ABORT_EARLY; // preamble/sync missing
+    }
+    if (b[0] != 0xaa || b[1] != 0xaa || b[2] != 0xaa || b[3] != 0xd3 || b[4] != 0x91 || b[5] != 0xd3 || b[6] != 0x91)
+        return DECODE_ABORT_EARLY; // preamble/sync missing
+
+    int totallength = bitbuffer->bits_per_row[0];
+    int packetsize  = (int)b[7];
+
+    // packet actually starts at b[8]
+    uint32_t unknown_field = ((uint32_t)b[8] << 24) | (b[9] << 16) | (b[10] << 8) | (b[11]); // not sure what these four bytes are.
+
+    // get xmitted pin (byte 12, inverted, reversed, bottom nibble)
+    uint8_t pin_container          = b[12];
+    uint8_t inverted_pin_container = ~pin_container;                   // invert the pin
+    uint8_t reversed_pin           = reverse8(inverted_pin_container); // reverse the bits
+
+    // convert just the first four bits to binary string
+    char pin_string[5] = {'\0'};
+    snprintf(pin_string, sizeof(pin_string), "%d%d%d%d",
+            (reversed_pin & 0x80 ? 1 : 0),
+            (reversed_pin & 0x40 ? 1 : 0),
+            (reversed_pin & 0x20 ? 1 : 0),
+            (reversed_pin & 0x10 ? 1 : 0));
+
+    // button that was pushed
+    uint8_t button_id = b[13];
+
+    // get label
+    char button_string[50] = {'\0'};
+
+    switch (button_id) {
+    case BUTTON_ID_ONE:
+        snprintf(button_string, sizeof(button_string), "On/Off Channel 1");
+        break;
+    case BUTTON_ID_TWO:
+        snprintf(button_string, sizeof(button_string), "On/Off Channel 2");
+        break;
+    case BUTTON_ID_S:
+        snprintf(button_string, sizeof(button_string), "Color Sync");
+        break;
+    case BUTTON_ID_M:
+        snprintf(button_string, sizeof(button_string), "ON/OFF Control - M");
+        break;
+    default:
+        snprintf(button_string, sizeof(button_string), "Unknown");
+        break;
+    }
+
+    // grab CRCs
+    // commenting out the below two lines because we're not using this crc
+    // see above for description.
+    //
+    // however, i'd still like the code to be here in case we need it in the future.
+    // uint8_t baby_crc            = b[14];
+    // uint8_t baby_calculated_crc = crc8(&b[8], 6, 1, 1);
+
+    // total CRC-16
+    // this CRC represents the entire packet sent by the modem in the remote
+    uint16_t total_crc            = (b[15] << 8) | (b[16]);
+    uint16_t calculated_total_crt = crc16(&b[7], 8, 0x8005, 0xFFFF);
+
+    if (total_crc != calculated_total_crt) {
+        // parity fail.
+        return DECODE_FAIL_MIC;
+    }
+
+    /* clang-format off */
+    data = data_make(
+            "model",                  "",                         DATA_STRING, "SRSmith-SRS2CTX",
+            "id",                     "Id",                       DATA_STRING, pin_string, // technically peoples pins should be different on each remote
```

Comment:
this should be `reversed_pin` not `pin_string`, right?

Reply:
`pin_string` is correct in this case, imho, i'd rather output something that makes sense to whoever is working with this (e.g., have it match the dip switches on/off on the battery compartment of the remote

Reply:
Sorry, we really want simple numbers. Presentation (pretty printing) is not a task for rtl_433.
There are devices that have ID or channel code settings like A,B,C, or I/O, or X,- -- we want to always show a simple number to make database logging and such easier.
But you can add something like `decoder_logf(decoder, 2, __func__, "Pin setting is %c%c%c%c", pin0, pin1, pin2, pin3);` for debugging.

Reply:
understood.